### DISCUSSION TITLE
Boolean value in array  [is it possible?]

### DISCUSSION THREAD

**Discussion by kenkit** [nlohmann/json#3533] [Q&A] [answered]
Hi,
How would we create an array with random params like this 
params can be order differently depending on the user needs.
```json
{"jsonrpc": "1.0", "id":"curltest", "method": "importaddress", "params": ["myaddress", "testing", false] }
```

**Comment by kenkit** [comment]
This is how I was doing it but the last bool vector cannot work
```cpp
  std::tuple<std::vector<std::string>, std::vector<int>, std::vector<bool>> params;
  request["params"] = nlohmann::json::array();
  for (auto param : std::get<0>(parameters))
  {
    request["params"].push_back(param);
  }
  for (auto param : std::get<1>(parameters))
  {
    request["params"].push_back(param);
  }
  for (auto param : std::get<2>(parameters))
  {
    request["params"].push_back(param);///fails here
  }
```

**Comment by falbrechtskirchinger** [comment, marked as answer]
`std::vector<bool>::reference` is not a `bool &` as on might expect (see https://en.cppreference.com/w/cpp/container/vector_bool).

Either of the options below should work.
```cpp
  for (bool param : std::get<2>(parameters))
  {
    request["params"].push_back(param);
  }

  for (auto param : std::get<2>(parameters))
  {
    request["params"].push_back(static_cast<bool>(param));
  }
```

**Reply by nlohmann** [reply]
Would it make sense to teach the library how to automatically consume `std::vector<bool>::reference`?

**Reply by falbrechtskirchinger** [reply]
I looked into that earlier and didn't think it would be worth the effort.

**Reply by falbrechtskirchinger** [reply]
Ok, because you asked, I took a second look. It's not as bad as I thought. PR coming your way.

**Reply by nlohmann** [reply]
FYI: #3534 is merged.

**Reply by kenkit** [reply]
Nice, thank you.
